lib/pentagon/__init__.py

```diff
@@ -388,6 +403,44 @@ def __prepare_ansible_cfg_vars(self):
         }
         return self.__render_template(template_name, template_path, target, context)
 
+    def __prepare_vpn_cfg_vars(self):
+        self.__get_vpn_ami_id()
+        template_name = "env.yml.jinja"
+        template_path = "{}/default/resources/admin-environment".format(self._repository_directory)
+        target = "{}/default/resources/admin-environment/env.yml".format(self._repository_directory)
+        context = {
+            'admin_vpn_key': self._ssh_keys['admin_vpn'],
+            'vpn_ami_id': self._vpn_ami_id
+        }
+        return self.__render_template(template_name, template_path, target, context)
+
+    def __get_vpn_ami_id(self):
+
+        self._vpn_ami_id = self._vpn_ami_id_placeholder
+
+        if self.get_arg('configure_vpn'):
+            if self.get_arg('vpn_ami_id'):
+                self._vpn_ami_id = self.get_arg('vpn_ami_id')
+            elif \
+                    self._aws_access_key != self._aws_access_key_placeholder and \
+                    self._aws_secret_key != self._aws_secret_key_placeholder and \
+                    self._aws_default_region != self._aws_default_region_placeholder:
+
+                logging.info("Getting VPN ami-id from AWS")
+
+                # Set EC2 environ vars for boto3 to use
+                os.environ['AWS_ACCESS_KEY'] = self._aws_access_key
+                os.environ['AWS_SECRET_KEY'] = self._aws_secret_key
+                os.environ['AWS_DEFAULT_REGION'] = self._aws_default_region
+                try:
+                    client = boto3.client('ec2')
+                    images = client.describe_images(Owners=self._ami_owners,Filters=self._vpn_ami_filters)
+                    self._vpn_ami_id = images['Images'][-1]['ImageId']
+                except Exception, e:
+                    logging.error("Encountered \" {} \" getting ami-id. VPN not configured fully".format(e))
```

Comment:
Maybe not non-critical, but it seems if the user explicitly tries to find an AMI, and Pentagon doesn't, it shouldn't create more work for the user. It looks like the `docs/vpn.md` has info for the AMI, maybe put a note here that the info can be found there?